tests/unit/s2n_tls13_cert_verify_test.c

```diff
@@ -142,39 +143,48 @@ int run_tests(const struct s2n_tls13_cert_verify_test *test_case, s2n_mode verif
         uint64_t bytes_in_hash = 0;
         s2n_pkey_type pkey_type = { 0 };
 
+        struct s2n_connection *verifying_conn = NULL, *sending_conn = NULL;
+        EXPECT_NOT_NULL(verifying_conn = s2n_connection_new(verifier_mode));
+        verifying_conn->actual_protocol_version = S2N_TLS13;
+        EXPECT_NOT_NULL(sending_conn = s2n_connection_new(verifier_mode == S2N_CLIENT ? S2N_SERVER : S2N_CLIENT));
+
         EXPECT_SUCCESS(s2n_stuffer_alloc(&certificate_in, S2N_MAX_TEST_PEM_SIZE));
         EXPECT_SUCCESS(s2n_stuffer_alloc(&certificate_out, S2N_MAX_TEST_PEM_SIZE));
 
-        struct s2n_connection *verifying_conn = NULL;
-        EXPECT_NOT_NULL(verifying_conn = s2n_connection_new(verifier_mode));
-        verifying_conn->actual_protocol_version = S2N_TLS13;
+        EXPECT_SUCCESS(s2n_connection_set_config(sending_conn, config));
+        sending_conn->handshake_params.our_chain_and_key = cert_chain;
+        sending_conn->handshake_params.server_cert_sig_scheme = &sig_scheme;
+        sending_conn->handshake_params.client_cert_sig_scheme = &sig_scheme;
+        sending_conn->secure->cipher_suite = &s2n_tls13_aes_128_gcm_sha256;
+        sending_conn->actual_protocol_version = S2N_TLS13;
+
         EXPECT_SUCCESS(s2n_connection_set_config(verifying_conn, config));
-        verifying_conn->handshake_params.our_chain_and_key = cert_chain;
-        verifying_conn->handshake_params.server_cert_sig_scheme = &sig_scheme;
-        verifying_conn->handshake_params.client_cert_sig_scheme = &sig_scheme;
         verifying_conn->secure->cipher_suite = &s2n_tls13_aes_128_gcm_sha256;
 
         EXPECT_SUCCESS(s2n_blob_init(&b, (uint8_t *) cert_chain_pem, strlen(cert_chain_pem) + 1));
         EXPECT_SUCCESS(s2n_stuffer_write(&certificate_in, &b));
         EXPECT_SUCCESS(s2n_stuffer_certificate_from_pem(&certificate_in, &certificate_out));
 
+        /* Extract public key from certificate and set it for verifying connection */
```

Comment:
Okay nevermind, I was totally wrong. Apparently the important line here is s2n_asn1der_to_public_key_and_type() which is actually instantiating the public key for the verifying connection. This normally happens when the verifying connection receives the certificate message in the handshake. Since this test doesn't do the handshake, it has to manually set it the verifying connection's public key.